components/binding-usage-controller/docs/proposals/usage-kind.md

```diff
@@ -0,0 +1,94 @@
+# Binding Usage Target proposal
+
+## Overview
+The **UsageKind** allows dynamic (in the runtime) **Binding Usage Controller** configuration. It allows to define, what (and how) resources can be bound with **ServiceBinding**. Currently, only two resources can be bound: **Deployment** and **Function**. Adding a new resource requires changes in the Binding Usage Controller code. The goal is to make it possible only with configuration.
+
+## UsageKind 
+
+The idea is to have a new cluster wide custom resource, which defines bindable resource, for example
+```yaml
+apiVersion: servicecatalog.kyma.cx/v1alpha1
+kind: UsageKind
+metadata:
+   name: function
+spec:
+   # displayName is a human readable name of the target
+   displayName: Function
+   
+   # the type of resource (target), which is labelled by the controller, is specified as resource group, kind and version (all fields required)
+   # there is a plan to make version field optional in the future - the controller can find the preferred version
+   resource:
+     group: kubeless.io
+     kind: function
+     version: v1beta1
+     
+   # labelsPath specifies a resource field, which contains labels
+   labelsPath: spec.deployment.spec.template.metadata.labels
+```
+
+## Ui-api-layer
+
+UI needs two new endpoints, which returns:
+ * all kinds, which can be used with binding usage. It is just a list of all binding usage targets
+ * all resources with given kind. The result must be filtered out by **metadata.ownerReference** field: if the resource contains **metadata.ownerReference**, the user should not see such resource in the UI.
+
+## Binding Usage Controller
+
+Binding usage controller takes the value of spec.usedBy.kind (binding usage resource), looks for corresponding target, which contains information about kind of resource to be bound. The controller must handle resources even if specified labels field (labelsPath) does not exists.
```

Comment:
``* Binding usage controller takes the value of the `spec.usedBy.kind` binding usage resource, looks for the corresponding target which contains information about the kind of resource to be bound. The controller must handle resources even if the specified `labelsPath`label field does not exist.``

Reply:
taret -> usage kind